robot/basestation/static/js/elements/feed/navbar.js

```diff
@@ -86,151 +111,71 @@ $(document).ready(() => {
-                  sendRequest('Arm', 'ping', printErrToConsole)
-                } else {
-                  $('#toggle-arm-listener-btn')[0].checked = false
-                }
-              },
-              serialType
-            )
-          }
-        }
-      })
-    }
+  $("a[id^='mux-']").mouseup(e => {
+    let muxId = $(e.target).attr("id")
+    setMuxChannel(muxId)
   })
 
-  $('#mux-2').mouseup(function () {
-    // Science
+  // Set device channel
+  function setMuxChannel (muxId) {
+    deviceName = getDeviceNameByMuxID(muxId)
     if (isListenerOpen() && getCookie('serialType') == 'uart') {
-      appendToConsole('Don\'t change the mux channel while a listener is open!')
+      rosLog(ROSINFO, 'Don\'t change the mux channel while a listener is open!')
     } else {
-      requestMuxChannel('#mux-2', function (msgs) {
+      requestMuxChannel('#' + muxId, function (msgs) {
         printErrToConsole(msgs)
 
-        if (msgs[0] == true && window.location.pathname == '/science') {
-          console.log('Activating Science Listener Node')
-
-          let serialType = getCookie('serialType')
-
-          if (serialType == '') {
-            appendToConsole('Serial type not yet defined!')
-            return
-          }
-
-          // automating opening listener and sending MCU ping in UART mode
-          if (serialType == 'uart') {
-            requestTask(
-              SCIENCE_LISTENER_TASK,
-              STATUS_START,
-              function (msgs) {
-                if (msgs[0]) {
-                  $('#science-listener-btn')[0].checked = true
-                  // try pinging MCU
-                  wait(1000)
-                  sendRequest('Science', 'ping', printErrToConsole)
-                } else {
-                  $('#science-listener-btn')[0].checked = false
-                }
-              },
-              serialType
-            )
+        if (muxId != 'mux-3') {
+          if (msgs[0]) {
+            rosLog(ROSINFO, 'Activating ' + deviceName + ' Listener Node')
+
+            let serialType = getCookie('serialType')
+
+            if (serialType == '') {
+              rosLog(ROSINFO, 'Serial type not yet defined!')
+              return
+            }
+
+            // automating opening listener and sending MCU ping in UART mode
+            if (serialType == 'uart') {
+              requestTask(
+                ROVER_LISTENER_TASK,
+                STATUS_START,
+                function (msgs) {
+                  if (msgs[0]) {
+                    $('#toggle-rover-listener-btn')[0].checked = true
+                    // try pinging MCU
+                    wait(1000)
+                    sendRequest(deviceName, 'ping', printErrToConsole)
+                  } else {
+                    $('#toggle-rover-listener-btn')[0].checked = false
+                  }
+                },
+                serialType
+              )
+            }
           }
         }
       })
     }
```

Comment:
This seems a little funky. 
1. If we get to this block then that means we have some non-PDS device selected, but if that's true then why would we have uart enabled?
2. In case of Arm, Science, Rover it's always going to request rover listener to activate

I think it would be safest to just remove this activation of listeners within the setMuxChannel function.

Reply:
Makes sense, removed